lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/TransportFactory.java

```diff
@@ -82,5 +101,27 @@ Transport newInstance(final String hostAddress,
                         final int retryTimeout,
                         final TcpPortProvider tcpPortProvider);
 
+  /**
+   * Creates a transport.
+   *
+   * @param hostAddress     a host address
+   * @param port            a listening port
+   * @param clientStage     a transport client-side stage
+   * @param serverStage     a transport server-side stage
+   * @param numberOfTries   the number of retries for connection
+   * @param retryTimeout    retry timeout
+   * @param tcpPortProvider tcpPortProvider
+   * @param protocol        protocol to use
+   * @return transport
+   */
+  Transport newInstance(final String hostAddress,
+                        int port,
```

Comment:
You don't have to refactor it in another PR because it is a new code. Please just add `final` here.